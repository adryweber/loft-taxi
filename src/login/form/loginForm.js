import React from "react";
import Typography from "@material-ui/core/Typography";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { useStyles } from "../../shared/style";

export const LoginForm = ({ login }) => {
    const [fields, setFields] = React.useState({ email: "", password: "" });
    const [invalid, setInvalid] = React.useState(false);

    const classes = useStyles();

    const onSubmit = login => e => {
        e.preventDefault();
        setInvalid(!login(fields.email, fields.password));
    };

    const InvalidMessage = () => {
        if (!invalid) {
            return null;
        }
        return (
            <Typography id="textError" variant="body2" component="span">
                Неправильные логин и пароль.
            </Typography>
        );
    };

    const handleChange = e => {
        setInvalid(false);
        setFields({ ...fields, [e.target.name]: e.target.value });
    };

    return (
        <form id="formLogin" onSubmit={onSubmit(login)}>
            <InvalidMessage />
            <TextField
                InputLabelProps={{ shrink: true }}
                fullWidth
                label="Эл. почта"
                margin="normal"
                name="email"
                value={fields.email}
                onChange={handleChange}
            />
            <TextField
                InputLabelProps={{ shrink: true }}
                fullWidth
                label="Пароль"
                margin="normal"
                name="password"
                value={fields.password}
                onChange={handleChange}
            />
            <Button
                className={classes.formBottomButton}
                variant="contained"
                color="primary"
                fullWidth
                type="submit"
            >
                Login
            </Button>
        </form>
    );
};
