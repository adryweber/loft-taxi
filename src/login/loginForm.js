import React from "react";
// import PropTypes from "prop-types";
import { Consumer } from "../shared/auth-context";
import Typography from "@material-ui/core/Typography";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
// import { useStyles } from "../shared/style";

// const classes = useStyles();
// console.log(classes);

const InvalidMessage = ({ invalid }) => {
    if (!invalid) {
        return null;
    }
    return (
        <Typography id="textError" variant="body2" component="span">
            Неправильные логин и пароль.
        </Typography>
    );
};

class LoginForm extends React.Component {
    state = { email: "", password: "", invalid: false };

    onSubmit = login => e => {
        e.preventDefault();
        login(this.state.email, this.state.password);
        this.setState({ invalid: true });
    };

    handleChange = e => {
        this.setState({ invalid: false });
        this.setState({ [e.target.name]: e.target.value });
    };

    render() {
        return (
            <Consumer>
                {({ isLoggedIn, login }) => (
                    <form id="formLogin" onSubmit={this.onSubmit(login)}>
                        <InvalidMessage invalid={this.state.invalid} />
                        <TextField
                            InputLabelProps={{ shrink: true }}
                            fullWidth
                            label="Эл. почта"
                            margin="normal"
                            name="email"
                            value={this.state.email}
                            onChange={this.handleChange}
                        />
                        <TextField
                            InputLabelProps={{ shrink: true }}
                            fullWidth
                            label="Пароль"
                            margin="normal"
                            name="password"
                            value={this.state.password}
                            onChange={this.handleChange}
                        />
                        <Button
                            // className={classes.formBottomButton}
                            variant="contained"
                            color="primary"
                            fullWidth
                            type="submit"
                        >
                            Login
                        </Button>
                    </form>
                )}
            </Consumer>
        );
    }
}

export { LoginForm };
