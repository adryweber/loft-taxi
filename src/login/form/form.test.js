import React from "react";
import { LoginForm } from "./loginForm";
import renderer from "react-test-renderer";

it("Login Form renders correctly", () => {
    const tree = renderer
        .create(<LoginForm login={() => ({ preventDefault: () => {} })} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
