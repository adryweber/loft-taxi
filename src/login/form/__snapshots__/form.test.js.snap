// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form renders correctly 1`] = `
<form
  id="formLogin"
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
      data-shrink={true}
    >
      Эл. почта
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="email"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
      data-shrink={true}
    >
      Пароль
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="password"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formBottomButton-5 MuiButton-containedPrimary MuiButton-fullWidth"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Login
    </span>
  </button>
</form>
`;
